inference-engine/tests/functional/plugin/shared/src/behavior/ov_plugin/life_time.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <fstream>
+
+#include <ngraph_functions/subgraph_builders.hpp>
+#include <base/behavior_test_utils.hpp>
+#include "behavior/ov_plugin/life_time.hpp"
+
+namespace ov {
+namespace test {
+namespace behavior {
+std::string OVHoldersTest::getTestCaseName(testing::TestParamInfo<std::string> obj) {
+    return "targetDevice=" + obj.param;
+}
+
+void OVHoldersTest::SetUp() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+    targetDevice = this->GetParam();
+    deathTestStyle = ::testing::GTEST_FLAG(death_test_style);
+    if (deathTestStyle == "fast") {
+        ::testing::GTEST_FLAG(death_test_style) = "threadsafe";
+    }
+    function = ngraph::builder::subgraph::makeConvPoolRelu();
+}
+
+void OVHoldersTest::TearDown() {
+    ::testing::GTEST_FLAG(death_test_style) = deathTestStyle;
+}
+
+#define EXPECT_NO_CRASH(_statement) \
+EXPECT_EXIT(_statement; exit(0), testing::ExitedWithCode(0), "")
+
+void release_order_test(std::vector<std::size_t> order, const std::string &deviceName,
+                        std::shared_ptr<ngraph::Function> function) {
+    ov::AnyVector objects;
+    {
+        runtime::Core core = createCoreWithTemplate();
+        auto compiled_model = core.compile_model(function, deviceName);
+        auto request = compiled_model.create_infer_request();
+        std::vector<runtime::VariableState> states;
+        try {
+            states = request.query_state();
+        } catch(...) {}
+        auto tensor = request.get_input_tensor();
+        auto any = compiled_model.get_metric(METRIC_KEY(SUPPORTED_METRICS));
+        // WA: Could not place into ov::Any due to implicit cast operator
+        std::shared_ptr<runtime::RemoteContext> ctx;
+        try {
+            ctx = std::make_shared<runtime::RemoteContext>(compiled_model.get_context());
+        } catch(...) {}
+
+        objects = {core, compiled_model, request, states, tensor, any, ctx};
+    }
+    for (auto&& i : order) {
+        objects.at(i) = {};
+    }
+}
+
+TEST_P(OVHoldersTest, Orders) {
+    std::vector<std::size_t> order(6);
+    std::iota(order.begin(), order.end(), 0);
```

Comment:
Shouldn't it be 1 here as incrementation value?

Reply:
It is the first value in range

Reply:
My miss, sorry